timesketch/frontend/src/components/Explore/TimelineChip.vue

```diff
@@ -14,28 +14,320 @@ See the License for the specific language governing permissions and
+                  <strong style="font-size:1.2rem; margin-bottom:10px;">Error detail</strong>
+                  <pre style="margin-top:10px;">{{ datasource.error_message }}</pre>
+                </div>
+              </b-message>
+            </div>
+          </div>
+        </div>
+        <button class="modal-close is-large" aria-label="close" @click="showInfoModal = !showInfoModal"></button>
+      </b-modal>
+
+      <!-- Timeline rename modal -->
+    <b-modal :active.sync="showEditModal" :width="640" scroll="keep">
+      <div class="modal-background"></div>
+      <div class="modal-content">
+        <div class="card">
+          <header class="card-header">
+            <p class="card-header-title">Rename timeline</p>
+          </header>
+          <div class="card-content">
+            <div class="content">
+              <form @submit.prevent> <!-- Without prevent(), the page will refresh -->
+                <div class="field">
+                  <div class="control">
+                    <input v-model="newTimelineName" class="input" type="text" required autofocus />
+                  </div>
+                </div>
+                <div class="field">
+                  <div class="control">
+                    <input class="button is-success" @click="rename()" type="submit" value="Save" />
+                  </div>
+                </div>
+              </form>
+            </div>
+          </div>
+        </div>
+      </div>
+      <button class="modal-close is-large" aria-label="close" @click="showEditModal = !showEditModal"></button>
+    </b-modal>
+
+    <!-- Analyzer logs modal -->
+      <b-modal :active.sync="showAnalyzerModal" :width="1024" scroll="keep">
+        <div class="modal-background"></div>
+        <div class="modal-content">
+          <div class="card">
+            <header class="card-header">
+              <p class="card-header-title">Analyzer logs for {{ timeline.name }}</p>
+            </header>
+            <div class="card-content" v-if="showAnalyzerModal">
+              <ts-analyzer-history :timeline="timeline" @closeHistory="showAnalyzerModal = false"></ts-analyzer-history>
+            </div>
+          </div>
+        </div>
+        <button class="modal-close is-large" aria-label="close" @click="showAnalyzerModal = !showAnalyzerModal"></button>
+      </b-modal>
+
+    </span>
+
+    <span
+      class="tag is-medium has-text-left"
+      style="cursor: pointer; margin-right: 7px;margin-bottom:7px;padding-right: 6px;"
+      :style="timelineColor(timeline)"
+      @click="toggleTimeline(timeline)"
+    >
+      {{ timeline.name }}
+      <!-- Show a warning icon, if there were import errors. -->
+      <span
+        v-if="datasourceErrors.length"
+        class="b-tooltips"
+        style="padding-right: 6px; padding-left: 12px;"
+        @click.stop
+      >
+        <b-tooltip :label="failedImportsMessage" :type="isDarkTheme ? 'is-dark' : 'is-light'">
```

Comment:
What about

```suggestion
        <b-tooltip :label="datasourceErrors.length + ' failed imports'" :type="isDarkTheme ? 'is-dark' : 'is-light'">
```
 
and get rid of the computed property below?

Reply:
Funny because I converted the label into a computed property thinking that it would be more readable. I'm fine either way, and since you suggested it, I'll change it back.